clerk/middleware_v2.go

```diff
@@ -73,6 +73,13 @@ func WithSessionV2(client Client) func(handler http.Handler) http.Handler {
 				return
 			}
 
+			// In development or staging environment and probably not a web browser request based on User Agent
+			if isDevelopmentOrStaging(client) && !strings.HasPrefix(r.UserAgent(), "Mozilla/") {
```

Comment:
I think we should explain *why* we do this instead of *how* here, considering it's a change in the core AuthV2 algorithm.

Reply:
Added a brief explanation